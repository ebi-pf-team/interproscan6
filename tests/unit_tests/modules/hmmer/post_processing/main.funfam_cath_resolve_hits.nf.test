nextflow_process {

    name "Test Process FUNFAM_CATH_RESOLVE_HITS"
    script "interproscan/modules/hmmer/post_processing/main.nf"
    process "FUNFAM_CATH_RESOLVE_HITS"

    stage {
        symlink "tests/unit_tests/test_inputs/hmmer_post_processing/cath/hmmer_out_file_1"
        symlink "tests/unit_tests/test_inputs/hmmer_post_processing/cath/hmmer_out_file_2"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = [
                    file("meta/tests/unit_tests/test_inputs/hmmer_post_processing/cath/hmmer_out_file_1"),
                    file("meta/tests/unit_tests/test_inputs/hmmer_post_processing/cath/hmmer_out_file_2"),
                ]
                input[1] = [
                    "--min-dc-hmm-coverage=80 --worst-permissible-bitscore 25 --output-hmmer-aln"
                ]
                input[2] = true
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
