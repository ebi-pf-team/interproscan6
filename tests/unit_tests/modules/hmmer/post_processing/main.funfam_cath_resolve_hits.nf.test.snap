{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "funfam.cath.resolved.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-23T11:53:05.734237"
    }
}